### PR TITLE
Add logging (first tier)

### PR DESCRIPTION
**Pull request by kpp**
# Description
Add tracing::instrument to get more logs in a structured way

## Linked Issues
- Related to https://github.com/chainwayxyz/citrea/issues/509

### CODE REVIEW

File: crates/sovereign-sdk/full-node/sov-prover-storage-manager/src/lib.rs [hunk 1]
```diff
@@ -281,7 +282,7 @@ where
         &mut self,
         block_header: &Da::BlockHeader,
     ) -> anyhow::Result<Self::NativeStorage> {
-        tracing::trace!("Requested native storage for block {:?} ", block_header);
+        trace!("Requested native storage for block {:?} ", block_header);
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-prover-storage-manager/src/lib.rs:285]: instead of printing the whole header, we can use its hash

Comment (eyusufatik) [reply crates/sovereign-sdk/full-node/sov-prover-storage-manager/src/lib.rs:285]: this applies to other logs in this file

File: crates/bitcoin-da/src/helpers/builders.rs [hunk 2]
```diff
@@ -184,9 +187,16 @@ fn build_commit_transaction(
     } else {
         (None, utxos)
     };
+    let mut iteration = 0;
     let mut last_size = size;
 
     let tx = loop {
+        if iteration % 100 == 0 {
+            trace!(iteration, "Trying to find commitment size");
+            if iteration > 1000 {
```
Comment (eyusufatik) [inline crates/bitcoin-da/src/helpers/builders.rs:196]: 1000 is too low for this imo

go for 5k maybe? 10k?

File: crates/bitcoin-da/src/service.rs [hunk 3]
```diff
@@ -104,8 +107,11 @@ impl BitcoinService {
                 // TODO find last tx by utxo chain
                 let mut prev_tx = None;
 
+                trace!("BitcoinDA queue is initialized. Waiting for the first request...");
+
                 // We execute commit and reveal txs one by one to chain them
                 while let Some(request) = rx.recv().await {
+                    trace!("A new request is received");
                     let fee_sat_per_vbyte = match this.get_fee_rate().await {
                         Ok(rate) => rate,
```
Comment (eyusufatik) [inline crates/bitcoin-da/src/service.rs:116]: wanna print the txid once the transaction is sent?

File: crates/ethereum-rpc/src/lib.rs
Comment (eyusufatik) [inline crates/ethereum-rpc/src/lib.rs:1]: can you log params in debug_trace* end sendRawTransaction endpoints?

File: crates/evm/src/evm/executor.rs [hunk 4]
```diff
@@ -96,17 +100,20 @@ pub(crate) fn execute_multiple_tx<
     let mut evm = CitreaEvm::new(db, block_env, config_env, ext);
 
     let mut tx_results = Vec::with_capacity(txs.len());
-    for tx in txs {
+    for (i, tx) in txs.iter().enumerate() {
+        let _span = trace_span!("Processing tx", i, signer = %tx.signer()).entered();
```
Comment (eyusufatik) [inline crates/evm/src/evm/executor.rs:104]: tx hash here would be useful

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/bitcoin-da/src/helpers/builders.rs (hunk 2 above)
Comment (kpp) [reply crates/bitcoin-da/src/helpers/builders.rs:196]: Sure

**Pull request merged**